packages/api/src/beacon/routes/events.ts

```diff
@@ -208,28 +208,28 @@ export function getTypeByEvent(config: ChainForkConfig): {[K in EventType]: Type
     ),
     [EventType.blobSidecar]: blobSidecarSSE,
 
-    [EventType.lightClientOptimisticUpdate]: {
-      toJson: (data) =>
-        getLightClientTypeFromHeader((data as unknown as allForks.LightClientOptimisticUpdate).attestedHeader)[
+    [EventType.lightClientOptimisticUpdate]: WithVersion(
+      (_, data) =>
+        getLightClientTypeFromHeader((data as allForks.LightClientOptimisticUpdate).attestedHeader)[
           "LightClientOptimisticUpdate"
-        ].toJson(data),
-      fromJson: (data) =>
+        ],
+      (_, data) =>
         getLightClientTypeFromHeader(
           // eslint-disable-next-line @typescript-eslint/naming-convention
           (data as {attested_header: allForks.LightClientHeader}).attested_header
-        )["LightClientOptimisticUpdate"].fromJson(data),
-    },
-    [EventType.lightClientFinalityUpdate]: {
-      toJson: (data) =>
-        getLightClientTypeFromHeader((data as unknown as allForks.LightClientFinalityUpdate).attestedHeader)[
+        )["LightClientOptimisticUpdate"]
+    ),
+    [EventType.lightClientFinalityUpdate]: WithVersion(
+      (_, data) =>
+        getLightClientTypeFromHeader((data as allForks.LightClientFinalityUpdate).attestedHeader)[
```

Comment:
The benefit of this having the version as part of the data is that we can now use the version from there instead of having to calculate it based on the attestedHeader

Reply:
Right, I can probably make the change for reasons of symmetry.

My understanding is that those events will be produced only, never consumed from external sources (as in as HTTP body payload). And the `version` info is useful for event consumers to properly handle them.
Both `from`/`to` JSON are required for technical reasons (e.g. are used for our tests). Out of curiosity, is there any scenario involving consumption of those events?

Reply:
> Out of curiosity, is there any scenario involving consumption of those events?

The light client uses these events, if nobody would consume them what would be the reason to produce them?

> never consumed from external sources (as in as HTTP body payload)

Those are not classical http requests / responses here as the eventstream uses [server-sent events](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events)

You can easily test this in the browser when running the node and calling `http://localhost:9596/eth/v1/events?topics=light_client_optimistic_update`

Reply:
My question was in the context of the lodestar codebase. As in, what triggers the `fromJSON` function. I understand lodestar light-client does?

Reply:
> My question was in the context of the lodestar codebase. As in, what triggers the `fromJSON` function. I understand lodestar light-client does?

See light client
https://github.com/ChainSafe/lodestar/blob/d7ca290d173dfa82b2977f3e94df6e5e8d12ef3e/packages/light-client/src/transport/rest.ts#L77-L91

Our http client library is generic and needs to be able to handle all responses